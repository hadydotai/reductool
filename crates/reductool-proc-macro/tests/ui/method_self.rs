use reductool_proc_macro::aitool;

struct S;

impl S {
    #[aitool]
    fn m(&self, x: i32) {}
}

fn main() {}
