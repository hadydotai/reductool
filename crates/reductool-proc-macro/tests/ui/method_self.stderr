error: #[aitool] must be placed on a free-standing function (no `self`).Move the function out of the `impl` block or remove the receiver.
 --> tests/ui/method_self.rs:7:10
  |
7 |     fn m(&self, x: i32) {}
  |          ^^^^^
